tests/tools/saunafs.sh

```diff
@@ -449,48 +449,75 @@ prepare_metalogger_() {
+#
+create_emulated_zoned_devices_for_cs_number_() {
+	if [[ ! $use_zoned_disks || ! $use_ramdisk ]]; then
+        return
+    fi
+
+	local chunkserver_index=${1:-chunkserver_id}
+	local n=${disks_per_chunkserver}
+	local zoned_prefix="zonefs:"
+	local firstDisk=$(echo "${chunkserver_index} * ${n}" | bc)
+	local lastDisk=$(echo "${firstDisk} + ${n} - 1" | bc)
+
+	for disk_number in $(seq -w ${firstDisk} ${lastDisk}); do
+		if mount | grep "sauna_nullb${disk_number}" &>/dev/null; then
+			sudo umount -l "/mnt/zoned/sauna_nullb${disk_number}" &>/dev/null
+		fi
 
-	if [[ $use_zoned_disks && $use_ramdisk ]]; then
-		zoned_prefix="zonefs:"
+		create_zoned_nullb 4096 $zone_size_mb $number_of_conv_zones \
+			$number_of_seq_zones ${disk_number}
 
-		local metadataPath
-		local dataPath
-		local devPath
-		local firstDisk=$(echo "${chunkserver_id} * ${n}" | bc)
-		local lastDisk=$(echo "${firstDisk} + ${n} - 1" | bc)
+		if [ -b /dev/sauna_nullb${disk_number} ]; then
+			local devPath="/dev/sauna_nullb${disk_number}"
+		elif [ -b /dev/nullb${disk_number} ]; then
+			local devPath="/dev/nullb${disk_number}"
+		fi
 
-		for disk_number in $(seq -w ${firstDisk} ${lastDisk}); do
-			if mount | grep "sauna_nullb${disk_number}" &>/dev/null; then
-				sudo umount -l "/mnt/zoned/sauna_nullb${disk_number}" &>/dev/null
-			fi
+		sudo mkzonefs -f -o perm=666 "${devPath}" &>/dev/null
 
-			create_zoned_nullb 4096 $zone_size_mb $number_of_conv_zones \
-				$number_of_seq_zones ${disk_number}
+		local metadataPath="/mnt/ramdisk/metadata/sauna_nullb${disk_number}"
+		local dataPath="/mnt/zoned/sauna_nullb${disk_number}"
 
-			if [ -b /dev/sauna_nullb${disk_number} ]; then
-				devPath="/dev/sauna_nullb${disk_number}"
-			elif [ -b /dev/nullb${disk_number} ]; then
-				devPath="/dev/nullb${disk_number}"
-			fi
+		if [ ! -d "${metadataPath}" ]; then
+			mkdir -pm 777 "${metadataPath}"
+		fi
 
-			sudo mkzonefs -f -o perm=666 "${devPath}" &>/dev/null
+		# Mount point
+		if [ ! -d "${dataPath}" ]; then
+			sudo mkdir -pm 777 "${dataPath}"
+		fi
 
-			metadataPath="/mnt/ramdisk/metadata/sauna_nullb${disk_number}"
-			dataPath="/mnt/zoned/sauna_nullb${disk_number}"
+		sudo mount -t zonefs "${devPath}" "${dataPath}"
+	done
+}
 
-			if [ ! -d "${metadataPath}" ]; then
-				mkdir -pm 777 "${metadataPath}"
-			fi
+create_sfshdd_cfg_() {
+	local n=$disks_per_chunkserver
+	local zoned_prefix=""
 
-			# Mount point
-			if [ ! -d "${dataPath}" ]; then
-				sudo mkdir -pm 777 "${dataPath}"
-			fi
+	if [[ $use_zoned_disks && $use_ramdisk ]]; then
+		zoned_prefix="zonefs:"
+		local firstDisk=$(echo "${chunkserver_id} * ${n}" | bc)
+		local lastDisk=$(echo "${firstDisk} + ${n} - 1" | bc)
 
-			sudo mount -t zonefs "${devPath}" "${dataPath}"
+		for disk_number in $(seq -w ${firstDisk} ${lastDisk}); do
+			local metadataPath="/mnt/ramdisk/metadata/sauna_nullb${disk_number}"
+			local dataPath="/mnt/zoned/sauna_nullb${disk_number}"
```

Comment:
Potential for bugs here, since if the paths differ from the function above, this may not be accurate information (e.g a later change changes the paths in the function above, but not here, or vice versa).

Reply:
Nice catch, thanks. Done.